roborazzi-compose/src/main/java/com/github/takahirom/roborazzi/RoborazziComposePreview.kt

```diff
@@ -0,0 +1,123 @@
+package com.github.takahirom.roborazzi
+
+import android.app.Activity
+import android.graphics.Color
+import android.view.ViewGroup
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewRootForTest
+import androidx.compose.ui.unit.dp
+import androidx.test.core.app.ActivityScenario
+import org.robolectric.Shadows
+import org.robolectric.shadows.ShadowDisplay
+import java.io.File
+import kotlin.math.roundToInt
+
+fun captureSizedRoboImage(
+  widthDp: Int,
+  heightDp: Int,
+  showBackground: Boolean,
+  backgroundColor: Long,
+  filePath: String,
+  roborazziOptions: RoborazziOptions = provideRoborazziContext().options,
+  content: @Composable () -> Unit,
+) {
+  if (!roborazziOptions.taskType.isEnabled()) return
+  registerRoborazziActivityToRobolectricIfNeeded(ComponentActivity::class.java)
+
+  val activityScenario = ActivityScenario.launch(ComponentActivity::class.java)
+
+  activityScenario.onActivity {
+    it.setShadowDisplay(
+      widthDp = widthDp,
+      heightDp = heightDp
+    )
+  }
+
+  activityScenario.use {
+    activityScenario.onActivity { activity ->
+
+      activity.setBackgroundColor(
+        showBackground = showBackground,
+        backgroundColor = backgroundColor
+      )
+      
+      activity.setContent(
+        content = content.withSize(widthDp = widthDp, heightDp = heightDp)
+      )
+
+      val composeView = activity.window.decorView
+        .findViewById<ViewGroup>(android.R.id.content)
+        .getChildAt(0) as ComposeView
+
+      val viewRootForTest = composeView.getChildAt(0) as ViewRootForTest
+      viewRootForTest.view.captureRoboImage(filePath, roborazziOptions)
+    }
+
+    // Closing the activity is necessary to prevent memory leaks.
+    // If multiple captureRoboImage calls occur in a single test,
+    // they can lead to an activity leak.
+  }
+}
```

Comment:
I am not sure where to put these methods `setBackgroundColor` and `setShadowDisplay`... unfortunately both are Activity dependent, so I cannot really put them together with the other ones inside `applyToRobolectricConfiguration()`

I am a bit concern about how to make users of that method aware that to successfully apply widthDp, heightDp, showBackground and backgroundColor, they need an Activity

Reply:
Thanks. I think it would be great if we could restore the color after the test. We can refer to @yschimke's implementation for this.

https://github.com/takahirom/roborazzi/pull/574/files#diff-dbee5836626b59835a193026779dc5c522d087620d2cbbad92ba72a15149ed1eR159

Reply:
Since the activity is closed after snapshoting, it should not be necessary